prov/sockets/src/sock_progress.c

```diff
@@ -1520,8 +1520,8 @@ static int sock_pe_process_rx_send(struct sock_pe *pe, struct sock_rx_ctx *rx_ct
 	if (!rx_entry->is_buffered &&
 	    (!(rx_entry->flags & FI_MULTI_RECV) ||
 	     (pe_entry->flags & FI_MULTI_RECV))) {
-		sock_rx_release_entry(rx_entry);
 		fastlock_acquire(&rx_ctx->lock);
+		sock_rx_release_entry(rx_entry);
```

Comment:
@shantonu - The state here is is_busy flag and rx_cancel checks for this.
We only need to make sure the state update (i.e., is_busy flag) and list update (dlist_remove in this case) should be protected with a lock.